packages/client/src/client4.ts

```diff
@@ -1928,9 +1928,9 @@ export default class Client4 {
         );
     };
 
-    getPostThread = (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false) => {
+    getPostThread = (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false, fromCreateAt?: number, fromPost?: string, direction?: 'up'|'down', perPage?: number) => {
```

Comment:
parameters with default values should always come after mandatory and before purely optional ones. The `perPage` and `direction` params can also get a default value (`PER_PAGE_DEFAULT` below and `down` which is the default for the API) and then the order would be as follows:

```suggestion
    getPostThread = (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false, direction: 'up'|'down' = 'down', perPage: number = PER_PAGE_DEFAULT, fromCreateAt?: number, fromPost?: string) => {
```

Reply:
The only thing I'd ask with that is that we have the parameters come in the same order if we have multiple functions that take similar parameters to avoid future confusion

Reply:
@michelengelen `perPage` is not a mandatory one, not on this PR at least.
I'll re-order nonetheless!

Reply:
no, they aren't ... that's correct